docs/features/network-security-controls/network-policy.md

```diff
@@ -58,7 +178,107 @@ spec:
   policyTypes:
   - Egress
 ```
-**OVN-Implementation:**
+
+#### **Selector Based Denial**
+
+We can selectively block all traffic from/to certain pods, by using the `podSelector` field: in this case blocking all unicast traffic from reaching apps labeled with demo.
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector:
+    matchLabels:
+      app: demo
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+#### **Selector Based filtering**
+
+The `from` for an ingress and the `to` for an egress can be used to form a list of rules that will be allowed as destinations or sources of data, as in the example below, which allows all incoming traffic from the namespace db_app into the networkPolicy namespace.
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-db-source
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          name: db_app
+```
+
+
+#### **CIDR Based Filtering**
+
+The `from` selector for an ingress and the `to` selector for an egress can also be used to specify specific CIDR ranges or `ip blocks`, allowing developers to create granular ACLs. In the example below, we allow all source connections from the 10.16.3.0/24 subnet that are communicating on the mySQL port tcp/3306
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: cidr-db-app
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 10.16.3.0/24
+        except:
+        - 10.16.3.1/32
+    ports:
+    - protocol: TCP
+      port: 3306
+```
+
+#### **Complex Filtering**
+
+Each egress/ingress rule is designed to be combined with other rules to create more complex filters. Note that NetworkPolicies follow a whitelist model, in which only the rules specified are allowed, everything else will be denied In this example the application allows ingress traffic from both the db_namespace and content servers located in the 10.16.24.0/24 subnet while only allowing egress through 173.23.32.4/32
```

Comment:
"in which only the rules specified are allowed, everything else will be denied"
only specified connections are allowed?
also a dot is missing in the end of that sentence.
"while only allowing egress through 173.23.32.4/32" - not sure what through means here, did you mean to?